vck/src/commonMain/kotlin/at/asitplus/wallet/lib/iso/DeviceAuth.kt

```diff
@@ -13,4 +16,13 @@ data class DeviceAuth(
     val deviceSignature: CoseSigned<ByteArray>? = null,
     @SerialName("deviceMac")
     val deviceMac: CoseSigned<ByteArray>? = null, // TODO is COSE_Mac0
-)
\ No newline at end of file
+) {
+
+    fun serialize() = vckCborSerializer.encodeToByteArray(this)
```

Comment:
Just as a discussion point: We have so many classes now that have a `serialize()` method and a `deserialize()` on their companion. Especially everything JWT **needs** those methods because the result is not a valid JSON.
This got me thinking: Do we want to introduce an Interface for that?

Reply:
If the interface doesn't introduce more lines of code and complexity, I'm fine withit.
I also think, that we introduced this methods to call them from Swift code, and it's mostly used as a shortcut internally nowadays.

Reply:
JWT needs these, other formats don't. Since it is kind of a convention, we could formalise it through an interface.
Most obvious issue with this idea: _naming things is hard_!